python/lsst/afw/geom/box.cc

```diff
@@ -147,8 +191,37 @@ PYBIND11_PLUGIN(_box) {
     clsBox2D.def("clip", &Box2D::clip);
     clsBox2D.def("getCorners", &Box2D::getCorners);
     clsBox2D.def("toString", &Box2D::toString);
+    clsBox2D.def(
+        "__repr__", [](Box2D const & self) {
+            return py::str("Box2D(minimum={}, dimensions={})").format(
+                py::repr(py::cast(self.getMin())),
+                py::repr(py::cast(self.getDimensions()))
+            );
+        }
+    );
+    clsBox2D.def(
+        "__str__", [](Box2D const & self) {
+            return py::str("(minimum={}, maximum={})").format(
+                py::str(py::cast(self.getMin())),
+                py::str(py::cast(self.getMax()))
+            );
+        }
+    );
```

Comment:
When moving the code I realized it could very confusing to no know whether one was looking at the dimensions of the box or its maximum point, so I added the kwarg-like syntax (which is now happily `eval`-able in the case of `__repr__`).  The rest of the changes are just the difference between the Python and C++ interfaces for string formatting.